### PR TITLE
chore(core): show the last pin digit for a while

### PR DESCRIPTION
**Pull request by bieleluk**
This PR:
- [`mercury`, `tt`] Introduces a feature, that shows the last typed PIN number for short period of time (1 second) before changing it to back to a dot icon.
- [`tr`] Changes the feature, where the last PIN digit is shown until non-timer (button) event to use aforementioned timeout.
~To unify the behaviour, it similarly changes the _show the entire PIN_  to be shown for 2 seconds instead of until any non-time (button) event.~
- [`tr`] ~Fixes overflowing PIN that does not fit to the screen.~ (done in the global layout implementation )
pin_overflow_digits

Changes include:

1. Adding ~`TimerToken`~ `Timer` (changed due to global layout) object to the keyboard struct.
2. (`tt`, `mercury`) Changing bool parameter  `display_digits` to a 3-state enum in the keyboard struct.
3. (`tt`, `mercury`) Extending `render_dots` function with parameter to show the last digit.
4. (`tr`) Changing any non-timer event handling with the introduced timer token event.
~5. (`tr`) Using timeout for showing entire PIN.~
~6. (`tr`) Showing cropped PIN string instead of full overflowing one.~

### CODE REVIEW

File: core/embed/rust/src/ui/model_tr/component/input_methods/pin.rs [hunk 1]
```diff
@@ -263,16 +270,20 @@ impl Component for PinEntry<'_> {
...
-            if self.show_last_digit {
-                self.show_last_digit = false;
-                self.update(ctx)
+        if !matches!(event, Event::Timer(_)) && self.show_real_pin {
+            self.show_real_pin = false;
+            self.update(ctx);
+        }
+
+        // Timeout for showing the last digit.
+        match event {
```
[the remark this answers is not in the document]
Comment (bieleluk) [reply core/embed/rust/src/ui/model_tr/component/input_methods/pin.rs:272]: It was possible just with one caveat that it is a non-time event
https://github.com/trezor/trezor-firmware/pull/4328/commits/32f357adee3b57fcfbaa172ad75755f5afa784eb

File: core/embed/rust/src/ui/model_tr/component/changing_text.rs [hunk 2]
```diff
@@ -177,7 +177,7 @@ impl ChangingTextLine {
         let x_offset = if self.text.len() % 2 == 0 { 0 } else { 2 };
 
         let baseline = Point::new(self.pad.area.x0 + x_offset, self.y_baseline());
-        shape::Text::new(baseline, self.text.as_ref())
+        shape::Text::new(baseline, text_to_display.as_ref())
```
[the remark this answers is not in the document]
Comment (bieleluk) [reply core/embed/rust/src/ui/model_tr/component/changing_text.rs]: This has been corrected in the global layout PR and is already in the `main` so it is not relevant anymore

**Review by obrusvit — APPROVED** [review]
LGTM.
- had one remaining comment regarding the code
- update `fixtures` (all langs)
- squash fixups

**Comment by bosomt** [comment]
**QA OK**

tested on  - Device: Trezor T3T1 2.8.6 regular (revision 688bddc887b8bb660640cc6e40a99036f2e87f1f)

**Pull request merged**